User create view improvements

TW80000 opened the pull request:
## Changes

### Main
* Hide a bunch of properties when creating a User
* Update client-side validation of a bunch of User properties
  * I used `validateObj` directly instead of `validationPredicates` because that way the back end doesn't do any validation checks. I don't want the back end to check validation because it breaks a large number of tests. I know this is bad, but we don't currently have a good solution for application-specific validation. I plan on thinking about this next, but consider it outside the scope of this PR.
* Made `PasswordView` the default view for `Password` properties
* Disable "Save" button in DAO Controller v2 if data is invalid
* Use properties supplied by `propertyWhitelist` directly so cloned properties with overrides are respected

### Other
* Formatting
* Hid a few props when creating Rules that don't make sense to set, such as `created` and `lastModified`

Commit:
Hide a bunch of User properties in CREATE controller mode

Commit:
Log errors to make debugging easier

Commit:
Formatting

Commit:
Improve visibility of a few Rule properties

Commit:
Make PasswordView the default view of Password properties

Commit:
Don't require organization

This was an application-specific requirement and isn't required in
general. Setting this to required breaks a number of tests due to
validation now failing. Need to think of a different way to make this
required for our application specifically.

Commit:
User validation updates

kgrgreer commented on src/foam/nanos/auth/User.js:
```diff
@@ -184,6 +184,11 @@ foam.CLASS({
       class: 'String',
       name: 'jobTitle',
       section: 'personal',
+      validateObj: function(jobTitle) {
```
What if we set this property through java or DIG?

kgrgreer commented on src/foam/nanos/auth/User.js:
```diff
@@ -373,14 +380,17 @@ foam.CLASS({
-          return 'Password must contain one lowercase letter, one uppercase letter, one digit, and be between 7 and 32 characters in length.';
+      // TODO: Validation that's consistent with the PasswordPolicy feature but
+      // also only applies on create.
+      validateObj: function(desiredPassword, passwordLastModified) {
```
What about Java/Swift/DIG?

Commit:
Revert validation changes

Commit:
Disable save button when data is not valid

Commit:
Show Address in user create mode

Commit:
Fix typo

Commit:
Use properties supplied by whitelist directly

This allows people to clone properties and override property properties
like label in propertyWhitelist.

TW80000 commented:
@kgrgreer please review again.

jlhughes commented on src/foam/nanos/auth/email/EmailVerificationWebAgent.java:
```diff
@@ -62,7 +63,10 @@ public void execute(X x) {
     } catch (Throwable t) {
-      message = "Problem verifying your email.<br>" + t.getMessage();
+      String msg = t.getMessage();
+      ((Logger) x.get("logger")).error(msg);
```
```suggestion
      ((Logger) x.get("logger")).error(msg, t);
```
wil generate a stack trace in the log file.

TW80000 replied:
@jlhughes do you want me to revert this then?

kgrgreer commented:
LGTM
